indico/modules/events/persons/controllers.py

```diff
@@ -347,3 +359,54 @@ def _process(self):
             tpl = get_template_module('events/persons/management/_person_list_row.html')
             return jsonify_data(html=tpl.render_person_row(person_data, bool(self.event.registration_forms)))
         return jsonify_form(form)
+
+
+class RHEventPersonSearch(RHPersonsBase):
+    def _search_event_persons(self, exact=False, **criteria):
+        criteria = {key: value.strip() for key, value in criteria.items() if value.strip()}
+
+        if not criteria:
+            return set()
+
+        query = EventPerson.query.distinct(EventPerson.id).filter(EventPerson.event_id == self.event.id)
+
+        query = (
+            query.filter(or_(
+                EventPerson.abstract_links.any(AbstractPersonLink.abstract.has(~Abstract.is_deleted)),
+                EventPerson.contribution_links.any(ContributionPersonLink.contribution.has(~Contribution.is_deleted)),
+                EventPerson.event_links.any(EventPersonLink.event.has(~Event.is_deleted)),
+                EventPerson.subcontribution_links.any(
+                    SubContributionPersonLink.subcontribution.has(~SubContribution.is_deleted)
+                ),
+                EventPerson.session_block_links.any())
+            )
+        )
+
+        for k, v in criteria.items():
+            query = query.filter(unaccent_match(getattr(EventPerson, k), v, exact))
+
+        # wrap as subquery so we can apply order regardless of distinct-by-id
+        query = query.from_self()
+        query = query.order_by(
+            db.func.lower(db.func.indico.indico_unaccent(EventPerson.first_name)),
+            db.func.lower(db.func.indico.indico_unaccent(EventPerson.last_name)),
+            EventPerson.id,
+        )
+        return query
+
+    @use_kwargs({
+        'first_name': fields.Str(validate=validate.Length(min=1)),
+        'last_name': fields.Str(validate=validate.Length(min=1)),
+        'email': fields.Str(validate=lambda s: len(s) > 3),
+        'affiliation': fields.Str(validate=validate.Length(min=1)),
+        'exact': fields.Bool(missing=False),
+    }, validate=validate_with_message(
+        lambda args: args.keys() & {'first_name', 'last_name', 'email', 'affiliation'},
+        'No criteria provided'
+    ), location='query')
+    def _process(self, exact, **criteria):
+        matches = self._search_event_persons(exact=exact, **criteria)
+        results = sorted((EventPersonSchema().dump(entry) for entry in matches), key=itemgetter('full_name'))
```

Comment:
Since you already sort the results in the query, you can use `EventPersonSchema().dump(matches[:10], many=True)` here.

Also do the slicing here; no need to serialize more results. Or even better, do something like this in `_search_event_persons`:

    return query.limit(10).all(), query.count()

That way you get the total count but only *retrieve* 10 rows. Not a big deal here since there aren't a ton of EventPersons even in bigger events, but no need to fetch objects from the DB if you never use them.